app/src/main/java/org/listenbrainz/android/viewmodel/BaseViewModel.kt

```diff
@@ -0,0 +1,30 @@
+package org.listenbrainz.android.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import org.listenbrainz.android.model.ResponseError
+
+abstract class BaseViewModel<UiState>: ViewModel() {
```

Comment:
abstract classes are a pain if we don't handle them properly. They end up being a waste of time in future. Just be sure to use it everywhere because it could end up biting you in the future. That being said, I think it fits well here rn and I don't see any downside of it right now so okay with it.

Reply:
I was thinking to delegate all these tasks of vm to a special vm only for this task app wide. What do you say?

Reply:
This sounds better and good to me. Separating vms is better than making all vms conform to the same methods

Reply:
Done